aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/MutationProcessorTest.java

```diff
@@ -50,31 +52,36 @@
     private MutationOutbox mutationOutbox;
     private AppSync appSync;
     private MutationProcessor mutationProcessor;
+    private ModelSchemaRegistry modelSchemaRegistry;
 
     /**
      * A {@link MutationProcessor} is being tested. To do so, we arrange mutations into
      * an {@link MutationOutbox}. Fake responses are returned from a mock {@link AppSync}.
+     * @throws AmplifyException while loading the modelSchemaRegistry
      */
     @Before
-    public void setup() {
+    public void setup() throws AmplifyException {
         LocalStorageAdapter localStorageAdapter = InMemoryStorageAdapter.create();
         this.synchronousStorageAdapter = SynchronousStorageAdapter.delegatingTo(localStorageAdapter);
         this.mutationOutbox = new PersistentMutationOutbox(localStorageAdapter);
         VersionRepository versionRepository = new VersionRepository(localStorageAdapter);
         Merger merger = new Merger(mutationOutbox, versionRepository, localStorageAdapter);
         this.appSync = mock(AppSync.class);
-        this.mutationProcessor = new MutationProcessor(merger, versionRepository, mutationOutbox, appSync);
+        this.modelSchemaRegistry = ModelSchemaRegistry.instance();
+        this.modelSchemaRegistry.load(ImmutableSet.of(BlogOwner.class));
+        this.mutationProcessor =
+            new MutationProcessor(merger, versionRepository, modelSchemaRegistry, mutationOutbox, appSync);
     }
 
     /**
      * Tests the {@link MutationProcessor#startDrainingMutationOutbox()}. After this method
      * is called, any content in the {@link MutationOutbox} should be published via the {@link AppSync}
      * and then removed.
-     * @throws DataStoreException On failure to interact with storage adapter during arrangement
+     * @throws AmplifyException On failure to interact with storage adapter during arrangement
```

Comment:
`AmplifyException` is thrown from `ModelSchema.fromModelClass(Class<? extends Model>)` which is in the core. I believe the only places where `DataStoreException` is expanded to `AmplifyException` is in the tests (and not in the Datastore APIs) and that is because instead of changing the test data from `Models` to `ModelSchema` I changed it in the tests requiring that exception to be thrown. `SubscriptionProcessorTest` is an example where `AmplifyException` was already being [thrown](https://github.com/aws-amplify/amplify-android/blob/fbef1e8f9ccfb25e140bbb5199d018b016499958/aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessorTest.java#L90)

I do see your point, let's talk about how we can avoid it in tests as well.

Reply:
Clarified offline, for now we are keeping it as is since the change is only in the unit tests.